### PR TITLE
Move quicksight monitoring to config.json and disable it in FE

### PR DESCRIPTION
**Pull request by dlpzx**
### Feature or Bugfix
- Bugfix

### Detail
Quicksight monitoring was enabled or disabled in the cdk.json, however it is more of a functional feature than an infrastructure configuration. Moving it to the config.json allows us to disable the monitoring tab in the frontend. This solves the bug reported in #374 

### Relates
- #374 

### Security
Please answer the questions below briefly where applicable, or write `N/A`. Based on
[OWASP 10](https://owasp.org/Top10/en/).

- Does this PR introduce or modify any input fields or queries - this includes
fetching data from storage outside the application (e.g. a database, an S3 bucket)?
  - Is the input sanitized?
  - What precautions are you taking before deserializing the data you consume?
  - Is injection prevented by parametrizing queries?
  - Have you ensured no `eval` or similar functions are used?
- Does this PR introduce any functionality or component that requires authorization?
  - How have you ensured it respects the existing AuthN/AuthZ mechanisms?
  - Are you logging failed auth attempts?
- Are you using or adding any cryptographic features?
  - Do you use a standard proven implementations?
  - Are the used keys controlled by the customer? Where are they stored?
- Are you introducing any new policies/roles/users?
  - Have you used the least-privilege principle? How?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by dlpzx** [comment]
I tested the frontend locally, but I have not tested the deployment. That's why it is still in draft

**Comment by noah-paige** [comment]
I like the way you changed Monitoring tab to fix this error...

But a general question - do we have any rule of thumb for what should be included in `cdk.json`  vs. `config.json` ?

File: frontend/src/modules/Administration/views/AdministrationView.js [hunk 1]
```diff
@@ -16,11 +16,13 @@ import { ChevronRightIcon, useSettings } from 'design';
...
 import { isModuleEnabled, ModuleNames } from 'utils';
+import config from '../../../generated/config.json';
 
-const tabs = [
-  { label: 'Teams', value: 'teams' },
-  { label: 'Monitoring', value: 'dashboard' }
-];
+const tabs = [{ label: 'Teams', value: 'teams' }];
+
+if (config.core.enable_quicksight_monitoring) {
```
Comment (noah-paige) [inline frontend/src/modules/Administration/views/AdministrationView.js:23]: should it be `config.core.features.enable_quicksight_monitoring`

File: deploy/stacks/param_store_stack.py [hunk 2]
```diff
@@ -146,6 +135,21 @@ def __init__(
                 ),
             )
 
+    @run_if(['core.enable_quicksight_monitoring'])
```
Comment (noah-paige) [inline deploy/stacks/param_store_stack.py:138]: Should it be `core.features.enable_quicksight_monitoring` ?

File: deploy/stacks/backend_stack.py [hunk 3]
```diff
@@ -436,6 +363,80 @@ def __init__(
                 internet_facing=internet_facing,
             )
 
+    @run_if(['core.enable_quicksight_monitoring'])
```
Comment (noah-paige) [inline deploy/stacks/backend_stack.py:366]: Should it be `core.features.enable_quicksight_monitoring` ?

**Review by noah-paige — APPROVED** [review]
lgtm - approving

**Comment by dlpzx** [comment]
> I like the way you changed Monitoring tab to fix this error...
> 
> But a general question - do we have any rule of thumb for what should be included in `cdk.json` vs. `config.json` ?

That's the elephant in the room. In principle I would define `cdk.json` parameters as infrastructure parameters used in the deployment of data.all application infra (CICD+FE+BE).  `config.json` parameters are application features' parameters used at runtime in the FE and BE. Most parameters fall into one of those 2 definitions, but we have some exceptions of config.json runtime parameters that affect the infra deployed. We cannot access cdk.json at runtime, (and it is very weird) that's why so far if a parameter needs to be in the config.json but also affects the infra (cdk.json) I have defined them in the config.json. In short:
- parameter that affects infra = cdk.json
- parameter that affects a feature at runtime = config.json
- parameter that affects both = config.json

**Pull request merged**